networkx/algorithms/minors/__init__.py

```diff
@@ -0,0 +1,30 @@
+"""
+Subpackages related to graph-minor problems.
+
+In graph theory, an undirected graph H is called a minor of the graph G if H
+can be formed from G by deleting edges and vertices and by contracting edges
+[1]_.
+
+References
+----------
+.. [1] https://en.wikipedia.org/wiki/Graph_minor
+"""
+
+from networkx.algorithms.minors import contraction
```

Comment:
I guess my only remaining question is: do we want to add this module to the top-level namespace, or only the functions within it? I.e. do we ever want users to do `from networkx import contraction` or `nx.contraction.contracted_edge` instead of calling the functions from directly from `nx`.

It seems superfluous to expose the module from `nx` when all of the relevant functions are already available from there, but I don't feel strong about it either way - just curious what others think.

Reply:
Thanks for bringing that up. We really need to make a consistent system and shift to that.  
I don't care much which way we go so long as most functions are available easily (usually that means at the top level of the namespace).  I'm fine to follow protocols from other Scientific Python packages too.  

What makes it easy for developers down the road. If someone later adds a function that uses a private function, will we remember to put the module into the namespace so users can get at the private function?  I'm hoping lazy imports will make it all much cleaner -- but I haven't thought enough about how that would impact cases like this.

Reply:
I tend to like having the module available in the namespace. It at least discourages people from giving the module and one of its attributes the same name (e.g. `datetime.datetime`). I also sometimes like being able to write the explicit long name (without having to do the explicit import beforehand). I don't have a strong opinion though.

I'd vote to leave it as-is.

Lazy imports doesn't solve the problem, both conventions (expose submodule names / only expose attributes) work just fine with lazy imports. (Using mkinit can control the convention by specifying specifying  `--noattrs` or `--nomods`, also `--lazy` argument should exist in the pypi version)

Reply:
A counterargument to adding an extra namespace for functions that are already in the main namespace is (to paraphrase PEP20) "there should be one (and preferably only one) way to do things". From this perspective, it's suboptimal to have two different (public) interfaces to identical functionality (e.g. `nx.contracted_edges` and `nx.contraction.contracted_edges`). 

In practice I don't think this would matter to users much as most are likely to use the name that requires the least amount of typing. However, I do think it is confusing in that there *are* namespaces in networkx (e.g. `bipartite`) that contain functions with the same names as those in the main namespace, but with *different behavior* - e.g. `nx.betweenness_centrality` vs. `nx.bipartite.betweenness_centrality`.

Anyways this is all part of a wider discussion that IMO probably needs an NXEP. For the particular instance in this PR, I would vote to not make `nx.contraction` publicly available until we have the wider policy on namespaces settled - it's much easier to add a namespace later on than to take one away!